internal/handlers/pg/msg_renamecollection.go

```diff
@@ -0,0 +1,154 @@
+		return nil, err
+	}
+
+	sourceNamespace, err := common.GetRequiredParam[string](document, document.Command())
+	if err != nil {
+		return nil, err
+	}
+
+	sourceDB, collection, err := extractFromNamespace(sourceNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	targetField, err := document.Get("to")
+	if err != nil {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrMissingField,
+			"BSON field 'renameCollection.to' is missing but a required field",
+			document.Command(),
+		)
+	}
+
+	targetNamespace, ok := targetField.(string)
+	if !ok {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			"BSON field 'renameCollection.to' is the wrong type 'bool', expected type 'string'",
+			document.Command(),
+		)
+	}
+
+	// we assume we cannot move a collection between databases, yet.
+	targetDB, targetCollection, err := extractFromNamespace(targetNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	if targetDB != sourceDB {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrNotImplemented,
+			"Can't rename a collection to another database",
+			document.Command(),
+		)
+	}
+
+	if sourceNamespace == targetNamespace {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrIllegalOperation,
+			"Can't rename a collection to itself",
+			document.Command(),
+		)
+	}
+
+	err = dbPool.InTransactionRetry(ctx, func(tx pgx.Tx) error {
+		return pgdb.RenameCollection(ctx, tx, sourceDB, collection, targetCollection)
+	})
+
+	switch {
+	case err == nil:
+		// do nothing
+	case errors.Is(err, pgdb.ErrAlreadyExist):
+		return nil, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrNamespaceExists,
+			"target namespace exists",
+		)
+	case errors.Is(err, pgdb.ErrTableNotExist):
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrNamespaceNotFound,
+			fmt.Sprintf("Source collection %s does not exist", sourceNamespace),
+			document.Command(),
+		)
+	case errors.Is(err, pgdb.ErrInvalidCollectionName):
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrIllegalOperation,
+			fmt.Sprintf(
+				"error with target namespace: Fully qualified namespace is too long. Namespace: %s Max: 255",
+				targetNamespace,
+			),
+			document.Command(),
+		)
+	}
+
+	var reply wire.OpMsg
+	must.NoError(reply.SetSections(wire.OpMsgSection{
+		Documents: []*types.Document{must.NotFail(types.NewDocument(
+			"ok", float64(1),
+		))},
+	}))
+
+	return &reply, nil
+}
+
+// extractFromNamespace returns the database and collection name from a given namespace.
+func extractFromNamespace(namespace string) (string, string, error) {
+	split := strings.Split(namespace, ".")
+
+	// we assume that the given namespace contains a single dot.
+	if len(split) != 2 {
+		return "", "", commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrInvalidNamespace,
+			"Invalid namespace specified "+namespace,
+		)
```

Comment:
Could you add a test case for this where namespace does not consist of `db.collection`, something like `invalid` should work.